### PR TITLE
Fixing order of type checking and value reassignment

### PR DESCRIPTION
**Pull request by mleah**
### Description

This is a PR to fix [an issue](https://github.com/detroit-labs/lineage-frontend/issues/992) we are seeing on the MetricsOne frontend.

This issue is currently up in a [PR](https://github.com/alexkuz/redux-devtools-inspector/pull/72) to be fixed on the `redux-devtools-inspector`  repo.  However, it has been up since May, and there is no resolution in sight.

[There is also an issue up for the original repo.](https://github.com/alexkuz/redux-devtools-inspector/issues/73)

### Testing Steps

In order to test this, you will need to pull this branch on `redux-devtools-inspector` into your FE.

1.  While on the FE, run `yarn remove redux-devtools-inspector`.
2.  After the command successfully finishes, run ` yarn add github:detroit-labs/redux-devtools-inspector#LNG-2305-Fixing-typechecking-order-issue`
3.  Once this command successfully finished, start up the FE and the MW locally.
4.  Ensure that the FE compiles correctly.
5.  Click around through the FE to make sure the app works.
6.  If you were not having the issue happen, this is the full test.  Please see me for more info if you have questions..
7.  If you were having the issue, this should resolve it 🤞 .

Once this PR is approved and merged in, I will put up a corresponding FE PR.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,5 @@
 node_modules
 npm-debug.log
 .DS_Store
-lib
```
Comment (mleah) [inline .gitignore:4]: Removed the lib from gitignore.  There are build issues when trying to pull this into the FE repo .  The lib isn't being properly built when being included in the FE, so including the lib here is a quick fix for this issue.

File: src/types.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// @noflow
```
Comment (mleah) [inline src/types.js:1]: Flow tests for this file were failing, even on master of the fork 😕 

This addition stops flow (a typing library) from trying to type this library.

File: src/tabs/getItemString.js [hunk 3]
```diff
@@ -5,12 +5,13 @@ import getType from '../utils/getType';
 import type { StylingFunction } from 'react-base16-styling';
 
 function getShortTypeString(val: any, diff?: boolean): string {
-  const type = getType(val);
 
   if (diff && Array.isArray(val)) {
     val = val[val.length === 2 ? 1 : 0];
   }
 
+  const type = getType(val);
```
Comment (mleah) [inline src/tabs/getItemString.js:13]: This is the change needed to (hopefully) fully fix the issue with the FE.

**Review by SlaunchaMan — APPROVED** [review]
LGTM! @hoffmanktz, this is ready for QA!

**Comment by hoffmanktz** [comment]
Not throwing any errors in the console and everything is working as expected. LGTM!

**Pull request merged**